float32 x
float32 y
float32 z
int64[] digital_input